File: Source/MKL/examples/vmlc/source/vdpacki.c
#include <stdio.h>
#include <math.h>
#include "func_interv.h"

#include "mkl_vml.h"

int main()
{
  double dA[VEC_LEN],dB1[VEC_LEN];

  int i=0,vec_len=VEC_LEN,incra=3;

  for(i=0;i<vec_len;i++) {
    dA[i]=(double)i+1;
    dB1[i]=0.0;
  }

  vec_len=vec_len/incra+1;
#if defined(MKL_VML_MIXED)
  vdPackI(vec_len,dA,incra,dB1);
#elif defined(MKL_VML_LOWER)
  vdpacki(&vec_len,dA,&incra,dB1);
#elif defined(MKL_VML_UPPER)
  VDPACKI(&vec_len,dA,&incra,dB1);
#else
# error You must define MKL_VML_CDECL for CDECL call convention or MKL_VML_STDCALL for STDCALL call convention.
#endif


  printf("vdPackI test/example programm\n\n");
  printf("           Before packing             After packing\n");
  printf("===============================================================================\n");
  for(i=0;i<VEC_LEN;i++) {
    printf("% 25.14f ",dA[i]);
    printf("% 25.14f",dB1[i]);
    printf("\n");
  }
}

File: Source/MKL/examples/vmlf/source/vdpacki.f
	programm MKL_VML_TEST

	include "func_interv.fi"

	real*8 dA(VEC_LEN)
	real*8 dB1(VEC_LEN)
	integer i,incra,len

	i=0
	incra=3

	do i=1,VEC_LEN
		dA(i)=i/1.0
		dB1(i)=0.0
	end do

	len=VEC_LEN/incra+1
	call VDPACKI(len,dA,incra,dB1)


	print *,"vdPackI test/example programm"
	print *,""
	print *,""

	print *,"           Before packing             After packing"
	print *,"==============================================================================="
	do i=1,VEC_LEN
		print 10,dA(i),dB1(i)
	end do

10	format (F25.13,F25.13)	

	end
